apps/tangle-dapp/app/claim/EligibleSection.tsx

```diff
@@ -0,0 +1,308 @@
+'use client';
+
+import '@webb-tools/tangle-substrate-types';
+
+import type { SubmittableExtrinsic } from '@polkadot/api/types';
+import type { ISubmittableResult } from '@polkadot/types/types';
+import { hexToU8a, stringToU8a, u8aToString } from '@polkadot/util';
+import {
+  decodeAddress,
+  isEthereumAddress,
+  keccakAsHex,
+} from '@polkadot/util-crypto';
+import { useConnectWallet } from '@webb-tools/api-provider-environment/ConnectWallet';
+import { useWebContext } from '@webb-tools/api-provider-environment/webb-context';
+import { PresetTypedChainId } from '@webb-tools/dapp-types/ChainId';
+import isValidAddress from '@webb-tools/dapp-types/utils/isValidAddress';
+import { WebbError, WebbErrorCodes } from '@webb-tools/dapp-types/WebbError';
+import Button from '@webb-tools/webb-ui-components/components/buttons/Button';
+import { useWebbUI } from '@webb-tools/webb-ui-components/hooks/useWebbUI';
+import { Typography } from '@webb-tools/webb-ui-components/typography/Typography';
+import { shortenHex } from '@webb-tools/webb-ui-components/utils/shortenHex';
+import { shortenString } from '@webb-tools/webb-ui-components/utils/shortenString';
+import { type FC, useEffect, useMemo, useState } from 'react';
+import { isHex } from 'viem';
+
+import ClaimingAccountInput from '../../components/claims/ClaimingAccountInput';
+import ClaimRecipientInput from '../../components/claims/ClaimRecipientInput';
+import { getPolkadotApiPromise } from '../../constants/polkadot';
+import toAsciiHex from '../../utils/claims/toAsciiHex';
+import getStatement from '../../utils/getStatement';
+import type { ClaimInfoType } from './types';
+
+enum Step {
+  InputAddress,
+  Sign,
+  SendingTx,
+}
+
+const EligibleSection: FC<
+  ClaimInfoType & { onSuccess?: (hash: string) => void }
+> = ({ amount, isRegularStatement, onSuccess }) => {
+  const { activeAccount, activeApi, activeWallet } = useWebContext();
+  const { toggleModal } = useConnectWallet();
+  const { notificationApi } = useWebbUI();
+
+  const [recipient, setRecipient] = useState(activeAccount?.address ?? '');
+
+  const [recipientErrorMsg, setRecipientErrorMsg] = useState('');
+
+  const [step, setStep] = useState(Step.InputAddress);
+
+  // Validate recipient input address after 1s
+  useEffect(() => {
+    const timeout = setTimeout(() => {
+      if (recipient && !isValidAddress(recipient)) {
+        setRecipientErrorMsg('Invalid address');
+      } else {
+        setRecipientErrorMsg('');
+      }
+    }, 500);
```

Comment:
I utilized the `TextField` component within the `ClaimRecipientInput` because it offers more customization options than the `Input` component, especially in terms of styling.